File: app/src/main/java/com/mytlogos/enterprise/model/Episode.java
package com.mytlogos.enterprise.model;

import org.joda.time.DateTime;

public interface Episode {
    int getEpisodeId();

    float getProgress();

    int getPartId();

    int getTotalIndex();

    int getPartialIndex();

    DateTime getReadDate();

    boolean isSaved();
}

File: app/src/main/java/com/mytlogos/enterprise/model/Episode.kt
package com.mytlogos.enterprise.model

import org.joda.time.DateTime

data class Episode(
    val episodeId: Int,
    val progress: Float,
    val partId: Int,
    override val partialIndex: Int,
    override val totalIndex: Int,
    val readDate: DateTime?,
    val isSaved: Boolean
) : Indexable